### PR TITLE
feat: add project audit endpoint

### PR DESCRIPTION
**Pull request by Sineaggi**
Similar to #949 but specific to [project audit events](https://docs.gitlab.com/ee/api/audit_events.html#project-audit-events).

No previous issues for this exist.

The code was tested to verify it works.

### CODE REVIEW

File: gitlab/v4/objects/projects.py [hunk 1]
```diff
@@ -85,6 +87,17 @@ class GroupProjectManager(ListMixin, RESTManager):
     )
 
 
+class ProjectAudit(RESTObject):
+    _id_attr = "id"
+
+
+class ProjectAuditManager(ListMixin, RESTManager):
```
Comment (nejch) [inline gitlab/v4/objects/projects.py:94]: Based on the upstream docs, this endpoint supports both single + all events endpoint. We have a convenience `RetrieveMixin` for endpoitns that support both .get() and .list():

```suggestion
class ProjectAuditManager(RetrieveMixin, RESTManager):
```

**Review by nejch — CHANGES_REQUESTED** [review]
Thanks a lot for this @Sineaggi! Just one small comment.

BTW - It might make sense to separate this into its own module, `audit_events.py`. Looking at the upstream docs, we also have instance and group audit events, so this could be expanded at a later point, and look something like the variables module:

https://github.com/python-gitlab/python-gitlab/blob/master/gitlab/v4/objects/variables.py

EDIT: I realize now that the current instance-level audit events were put into `events.py`, but they seem to have their own section in gitlab's api, so moving them together into `audit_events.py` might still make sense.

Then import the project-specific classes in `projects.py`.

If you feel like it, you could also add some unit tests for this as a sanity check, using the example responses from your API link as mocks. Have a look at the existing unit tests: https://github.com/python-gitlab/python-gitlab/tree/master/gitlab/tests/objects

Thanks again!

**Comment by Sineaggi** [comment]
@nejch code moved to `audit_events.py`, moved to `RetrieveMixin`, added tests that execute both .get() and .list()

File: gitlab/tests/objects/test_audit_events.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        "target_type": "Project",
+        "target_details": "twitter/typeahead-js",
+        "ip_address": "127.0.0.1",
+        "entity_path": "twitter/typeahead-js",
+    },
+    "created_at": "2020-05-26T22:55:04.230Z",
+}
+
+audit_events_url = re.compile(
+    r"http://localhost/api/v4/(((groups|projects)/1)|(admin/ci))/audit_events"
```
Comment (nejch) [inline gitlab/tests/objects/test_audit_events.py:35]:
```suggestion
    r"http://localhost/api/v4/((groups|projects)/1/)audit_events"
```

File: gitlab/tests/objects/test_audit_events.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+)
+
+audit_events_url_id = re.compile(
+    rf"http://localhost/api/v4/(((groups|projects)/1)|(admin/ci))/audit_events/{id}"
```
Comment (nejch) [inline gitlab/tests/objects/test_audit_events.py:39]:
```suggestion
    rf"http://localhost/api/v4/((groups|projects)/1/)audit_events/{id}"
```

**Review by nejch — COMMENTED** [review]
Thanks again! Just found a little copy/paste leftover, I can push this.

**Review by nejch — APPROVED** [review]

**Pull request merged**